src/Clients/CacheResponse/Actions/CacheKeyMakerAction.php

```diff
@@ -4,18 +4,28 @@
 
 namespace StrictPhp\HttpClients\Clients\CacheResponse\Actions;
 
-use Psr\Http\Message\StreamInterface;
-use Psr\Http\Message\UriInterface;
+use Psr\Http\Message\RequestInterface;
 use StrictPhp\HttpClients\Contracts\CacheKeyMakerActionContract;
 
 final class CacheKeyMakerAction implements CacheKeyMakerActionContract
 {
-    public function execute(array &$headers, StreamInterface $body, UriInterface $uri): void
+    public function execute(RequestInterface $request): string
     {
+        $headers = $request->getHeaders();
+        $body = $request->getBody();
+
         foreach ($headers as $name => $header) {
             if (str_starts_with(strtolower($name), 'x-')) {
                 unset($headers[$name]);
             }
         }
+
+        return md5(serialize([
```

Comment:
Was wondering if serialize is the right call, I can see, reason is probably the`$headers`.

OpenAI reports that `json_encode` could be faster and more reproducible between environments -> https://chat.openai.com/share/df1da483-465a-49d6-80e1-fc6f21992b58